src/HealthGPS/kevin_hall_model.cpp

```diff
@@ -168,18 +221,23 @@ void KevinHallModel::set_nutrient_intakes(Person &person) const {
 }
 
 void KevinHallModel::initialise_energy_intake(Person &person) const {
-    set_energy_intake(person);
 
-    // Begin at steady state (EI = EE).
-    person.risk_factors["EnergyExpenditure"_id] = person.risk_factors.at("EnergyIntake"_id);
+    // Initialise energy intake.
+    set_energy_intake(person);
 
-    // TODO: set old value to new value
+    // Start with previous = current.
+    double energy_intake = person.risk_factors.at("EnergyIntake"_id);
+    person.risk_factors["EnergyIntake_previous"_id] = energy_intake;
 }
 
 void KevinHallModel::update_energy_intake(Person &person) const {
-    set_energy_intake(person);
 
-    // TODO: set old value to previous value
+    // Set previous energy intake.
+    double previous_energy_intake = person.risk_factors.at("EnergyIntake"_id);
+    person.risk_factors.at("EnergyIntake_previous"_id) = previous_energy_intake;
```

Comment:
Why do you need to save the previous value? (also in the previous function)

Reply:
This is to do with the computing the change in physical activity thing I mentioned in the previous PR: `double delta_EI = EI - EI_0`. Some parts of the Kevin Hall equations depend on previous risk factor values as well as current values.